cli/src/pcluster/config/update_policy.py

```diff
@@ -125,6 +134,68 @@ def is_slurm_queues_change(change):
     return any(path.startswith("SlurmQueues[") for path in change.path)
 
 
+def extract_type_and_name_from_path(path):
+    obj_type = re.search(".+(?=\\[)", path).group(0)
+    obj_name = re.search("(?<=\\[).+(?=\\])", path).group(0)
+    return obj_type, obj_name
+
+
+def get_q_from_config(change, config):
+    q_type, q_name = None, None
+    for path in change.path:
+        if re.search("Queues\\[", path):
+            q_type, q_name = extract_type_and_name_from_path(path)
+    if q_type and q_name:
+        for queue in config.get("Scheduling", {}).get(q_type, {}):
+            if queue.get("Name", None) == q_name:
+                return queue
+    return None
+
+
+def get_cr_from_config(change, config):
+    cr_type, cr_name = None, None
+    for path in change.path:
+        if re.search("ComputeResources\\[", path):
+            cr_type, cr_name = extract_type_and_name_from_path(path)
+    if cr_type and cr_name:
+        queue = get_q_from_config(change, config)
+        for compute_resource in queue.get(cr_type, {}):
+            if compute_resource.get("Name", None) == cr_name:
+                return compute_resource
+    return None
+
+
+def is_placement_group_managed_for_compute_resource(queue_networking, compute_resource_networking):
+    cr_pg = compute_resource_networking.get("PlacementGroup", None)
+    cr_pg_enabled = compute_resource_networking.get("PlacementGroup", {}).get("Enabled")
+    q_pg_enabled = queue_networking.get("PlacementGroup", {}).get("Enabled")
+    if (cr_pg is None and q_pg_enabled) or cr_pg_enabled:
+        return True
+    return False
+
+
+def is_managed_placement_group_deletion(change, patch):
+    base_q = get_q_from_config(change, patch.base_config)
+    base_cr = get_cr_from_config(change, patch.base_config)
+    target_q = get_q_from_config(change, patch.target_config)
+    target_cr = get_cr_from_config(change, patch.target_config)
+    if base_q and target_q and base_cr and target_cr:
+        if is_placement_group_managed_for_compute_resource(
+            base_q.get("Networking", {}), base_cr.get("Networking", {})
+        ) and not is_placement_group_managed_for_compute_resource(
+            target_q.get("Networking", {}), target_cr.get("Networking", {})
+        ):
+            print("return a")
+            return True
+    else:
+        if base_q.get("Networking", {}).get("PlacementGroup", {}).get("Enabled") and not target_q.get(
```

Comment:
We can copy some utility functions from https://github.com/aws/aws-parallelcluster/blob/develop/tests/integration-tests/utils.py#L416 to here

Reply:
Thanks for pointing that one out!  It's not quite the same as I am interested in the value as well is whether it exists, since it could be false, so I think I'll leave this as is, but I am going to take a closer look at those utility functions!